r2-core/src/main/java/com/linkedin/r2/transport/http/client/SmoothRateLimiter.java

```diff
@@ -179,9 +179,14 @@ public void setRate(double permitsPerPeriod, long periodMilliseconds, int burst)
     ArgumentUtil.checkArgument(permitsPerPeriod >= 0, "permitsPerPeriod");
     ArgumentUtil.checkArgument(periodMilliseconds > 0, "periodMilliseconds");
     ArgumentUtil.checkArgument(burst > 0, "burst");
-
-    _rate = new Rate(permitsPerPeriod, periodMilliseconds, burst);
-    _scheduler.execute(_eventLoop::updateWithNewRate);
+    Rate newRate = new Rate(permitsPerPeriod, periodMilliseconds, burst);
```

Comment:
Can we not creating a new Rate object when permitsPerPeriod and periodMilliseconds are equal to the same rate? (So instead of creating a new Rate object and using Object.eqauls, we just directly compare permitsPerPeriod and periodMilliseconds)